dask/dataframe/io/delta/core.py

```diff
@@ -0,0 +1,289 @@
+        API) will be used. If 'pyarrow-legacy' is specified, ArrowLegacyEngine
+        will be used (which leverages the pyarrow.parquet.ParquetDataset API).
+        NOTE: The 'pyarrow-legacy' option (ArrowLegacyEngine) is deprecated
+        for pyarrow>=5.
+    checkpoint: int, default None
+        DeltaTable protocol aggregates the json delta logs and creates parquet
+        checkpoint for every 10 versions.This will save some IO.
+        for example:
+            if you want to read 106th version of dataset, No need to read all
+            the all the 106 json files (which isIO heavy), instead read 100th
+            parquet checkpoint which contains all the logs of 100 versions in single
+            parquet file and now read the extra 6 json files to match the given
+            106th version.
+            (so IO reduced from 106 to 7)
+    storage_options : dict, default None
+        Key/value pairs to be passed on to the file-system backend, if any.
+    """
+
+    def __init__(
+        self,
+        path: str,
+        version: int = 0,
+        columns=None,
+        engine="auto",
+        checkpoint=None,
+        storage_options=None,
+    ):
+        self.path = str(path).rstrip("/")
+        self.version = version
+        self.pq_files = set()
+        self.delta_log_path = f"{self.path}/_delta_log"
+        self.fs, _, _ = get_fs_token_paths(path, storage_options=storage_options)
+        self.engine = engine
+        self.columns = columns
+        self.checkpoint = (
+            checkpoint if checkpoint is not None else self.get_checkpoint_id()
+        )
+        self.storage_options = storage_options
+        self.schema = None
+
+    def get_checkpoint_id(self):
+        """
+        if _last_checkpoint file exists, returns checkpoint_id else zero
+        """
+        try:
+            with self.fs.open(
+                f"{self.delta_log_path}/_last_checkpoint"
+            ) as last_checkpoint:
+                last_checkpoint_version = json.load(last_checkpoint)["version"]
+        except FileNotFoundError:
+            last_checkpoint_version = 0
+        return last_checkpoint_version
+
+    def get_pq_files_from_checkpoint_parquet(self):
+        """
+        use checkpoint_id to get logs from parquet files
+        """
+        if self.checkpoint == 0:
+            return
+        checkpoint_path = (
+            f"{self.delta_log_path}/{self.checkpoint:020}.checkpoint.parquet"
+        )
+        if not self.fs.exists(checkpoint_path):
+            raise ValueError(
+                f"Parquet file with the given checkpoint {self.checkpoint} does not exists: "
+                f"File {checkpoint_path} not found"
+            )
+
+        parquet_checkpoint = read_parquet(
+            checkpoint_path, engine=self.engine, storage_options=self.storage_options
+        ).compute()
+
+        # reason for this `if condition` was that FastParquetEngine seems to normalizes the json present in each column
```

Comment:
You. mean it's a structure-type column (not JSON)

Reply:
yeah that's right, fastparquet  was flattening the JSON schema, i.e let's say col1 has JSON  ({"a":1, "b":2}), Pyarrow is able to read a column of JSON, but Fastparquet reads more number of columns like `col1.a`, `col1.b`